x-pack/plugins/monitoring/server/kibana_monitoring/collectors/get_settings_collector.js

```diff
@@ -79,10 +79,16 @@ export function getSettingsCollector(server, kbnServer) {
       // remember the current email so that we can mark it as successful if the bulk does not error out
       shouldUseNull = !!defaultAdminEmail;
 
-      return {
-        kibana: getKibanaInfoForStats(server, kbnServer),
-        ...kibanaSettingsData
-      };
+      // return nothing when there was no result
+      let settingsDoc;
+      if (kibanaSettingsData != null) { // test null or undefined
```

Comment:
This won't catch undefined, but reading through the code it doesn't have to do that.

Reply:
++ I'd just update the comment to not say `or undefined`.

Reply:
Using double-equals to check for null or undefined is a habit I've acquired from working in Canvas:

```js
> var foo
> foo == null ? "hi" : "bye"
"hi"